src/Chapter2.hs

```diff
@@ -348,9 +350,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf = take =<< halfLen
+  where
+    halfLen = flip div 2 . length
```

Comment:
You can use a function as an operator if you put it in the backticks. So you can avoid `flip`:

```haskell
`div` 2
```

Reply:
I always forget about the infix notation lol, updated now 😄